src/main/kotlin/no/nav/familie/ks/sak/kjerne/endretutbetaling/EndretUtbetalingAndelService.kt

```diff
@@ -138,6 +144,40 @@ class EndretUtbetalingAndelService(
             it.copy(id = 0, behandlingId = behandling.id, erEksplisittAvslagPåSøknad = false)
         endretUtbetalingAndelRepository.save(kopiertOverEndretUtbetalingAndel)
     }
+
+    fun sanityBegrunnelseTilRestFormat(
+        sanityBegrunnelser: List<SanityBegrunnelse>,
+        begrunnelse: IBegrunnelse,
+    ): List<SanityBegrunnelseMedEndringsårsakResponseDto> {
+        val sanityBegrunnelse = begrunnelse.tilSanityBegrunnelse(sanityBegrunnelser)
+
+        if (sanityBegrunnelse != null) {
+            val visningsnavn = sanityBegrunnelse.navnISystem
+
+            return listOf(
+                SanityBegrunnelseMedEndringsårsakResponseDto(
+                    id = begrunnelse,
+                    navn = visningsnavn,
+                    endringsårsaker = sanityBegrunnelse.endringsårsaker,
+                ),
+            )
+        }
+
+        return emptyList()
+    }
+
+    fun sanityBegrunnelserMedEndringsårsak(sanityBegrunnelser: List<SanityBegrunnelse>) =
+        (NasjonalEllerFellesBegrunnelse.entries + EØSBegrunnelse.entries)
+            .groupBy { it.begrunnelseType }
+            .mapValues { begrunnelseGruppe ->
+                begrunnelseGruppe.value
+                    .flatMap { endretUtbetalingBegrunnelse ->
```

Comment:
```suggestion
                    .flatMap { begrunnelse ->
```

På dette punktet av koden så er det ikke gitt at begrunnelsen man har trukket ut fra NasjonalEllerFellesBegrunnelse.entries + EØSBegrunnelse.entries er en endretUtbetalingBegrunnelse

Reply:
Jepp, enig. Fikser det lokalt slik at jeg også får med samme renamingen til stedet den variabelen brukes 🙌